code/game/objects/items/storage/belt.dm

```diff
@@ -16,6 +16,20 @@
 	///TRUE Means that it closes a flap over its contents, and therefore update_icon should lift that flap when opened. If it doesn't have _half and _full iconstates, this doesn't matter either way.
 	var/flap = TRUE
 
+/obj/item/storage/belt/gun/flaregun/dump_into(obj/item/storage/Main, mob/user)
+
+	if(length(holstered_guns) < 1 && length(contents) >= (storage_slots-1))
+
+		to_chat(user, SPAN_WARNING("[src] is full."))
+		return FALSE
+	return ..()
+
+/obj/item/storage/belt/gun/flaregun/handle_item_insertion(obj/item/Warn, prevent_warning = 0, mob/user)
+
+	if(Warn.type == /obj/item/device/flashlight/flare && length(holstered_guns) < 1 && contents.len >= (storage_slots-1))
```

Comment:
Two things (I'd just make it a suggestion, but Warn is also requested to be changed):
- `length(contents)` instead of `contents.len` Faster, and handles null.
- `istype` instead of `.type ==` e.g. `istype(new_item, /obj/item/device/flashlight/flare)`
As I mentioned previously in a requested change, doing a .type comparision like this is effectively what the istypestrict macro is doing. We want to allow subtypes, so use istype.

Reply:
replace contents.len to what you propose.
change to istype like you proposed.